### DISCUSSION TITLE
Using the eforms-core-java-1.0.0-sources.jar library in an environment with JDK 1.8

### DISCUSSION THREAD

**Discussion by AlexAS81** [OP-TED/eForms-SDK#423] [Q&A] [answered]
We have a problem when using the eforms-core-java-1.0.0-sources.jar library in our application. Since the library is compiled with JDK 1.11 and our application is developed in JDK 1.8 and we have no option to install JDK 1.11.

Is there a way to use the library in an application with JDK 1.8?

Thanks in advance

**Comment by rousso** [comment, marked as answer]
Hi @AlexAS81,

Good point. Hard to say. To do this we need to change the sources and make sure the language features we use are backward compatible with Java 8. This of course has two side effects (apart from the fact that it requires some resources to implement):
- We will need to maintain the backward compatibility of the library which will limit our implementation options in the future
- What about the other libraries like the EFX Toolkit?

So, it is hard for me to say "yes" to this request. Did you try to change the sources to make them backwards compatible? If you did or you intend to, then you are welcome to make a pull request which we can review and consider. I cannot make any promisses however for the reasons I just explained.

Kind regards,
Yannis
